src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -164,22 +169,37 @@ void ProcessScope(LogRecordScope scope, OtlpLogs.LogRecord otlpLog)
             return otlpLogRecord;
         }
 
-        private static void AddStringAttribute(this RepeatedField<OtlpCommon.KeyValue> repeatedField, string key, string value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddAttribute(this OtlpLogs.LogRecord logRecord, OtlpCommon.KeyValue attribute, int maxAttributeCount)
         {
-            repeatedField.Add(new OtlpCommon.KeyValue
+            if (logRecord.Attributes.Count < maxAttributeCount)
```

Comment:
@alanwest One random thought looking at this PR about how we could improve things...

We build the `OtlpCommon.KeyValue` before we check if we will drop it. Could probably improve that so we skip the allocation(s) if we know we are up against the limit?

Reply:
@alanwest I just tried changing it for the LogRecord extension, but we would need something likely a dryrun for TryTransformTag, otherwise we would count attributes as dropped, that were ignored by TryTransformTag before.